app/src/org/commcare/preferences/FilePreferenceDialogFragmentCompat.java

```diff
@@ -56,6 +66,26 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent) {
         }
     }
 
+    private String getNewPathFromUri(Uri uri) {
+        // We can't get access to filePath but only the uri and
+        // since we are going to use this path later on in some other
+        // activity stack which will not have permissions for this uri
+        // we need to copy the file internally and pass the filepath of the new file
+        String filePath = getContext().getFilesDir().getAbsolutePath() + "/temp/" + FileUtil.getFileName(uri.toString());
+        File file = new File(filePath);
+        try {
+            InputStream fileStream = getContext().getContentResolver().openInputStream(uri);
+            FileUtil.ensureFilePathExists(file);
+            FileUtil.copyFile(fileStream, file);
+        } catch (IOException e1) {
+            e1.printStackTrace();
+            Logger.log(LogTypes.TYPE_MAINTENANCE, "Getting new file path from uri failed due to " + e1.getMessage());
+            Toast.makeText(getActivity(), Localization.get("file.selection.failed"), Toast.LENGTH_LONG).show();
+            return null;
```

Comment:
This seems like  kind of an odd return path to me v. throwing an I/O exception, since this is the fallback case. 

Do we really expect code to be able to function effectively if this method fails?